app/controllers/rankings_controller.rb

```diff
@@ -1,79 +1,123 @@
-#   # DELETE /rankings/1.json
-#   def destroy
-#     @ranking.destroy
-#     respond_to do |format|
-#       format.html { redirect_to rankings_url, notice: 'Ranking was successfully destroyed.' }
-#       format.json { head :no_content }
-#     end
-#   end
-#
+  before_action :set_fields, only: [:show, :edit, :create, :update, :end]
+  helper_method :valid_secret_id?
+
+  # GET /rankings
+  # GET /rankings.json
+  # def index
+  #   @rankings = Ranking.all
+  # end
+
+  # GET /rankings/1
+  # GET /rankings/1.json
+  def show
+  end
+
+  # GET /rankings/new
+  def new
+    @ranking = Ranking.new
+  end
+
+  def valid_secret_id?
+    params[:secret_id] == @participant.secret_id
+  end
+
+  # GET /rankings/1/edit
+  def edit
+    unless valid_secret_id?
+      flash[:message] = "Access denied."
+    end
+  end
+
+  def end
+
+  end
+
+  # POST /rankings
+  # POST /rankings.json
+  def create
+    # @ranking = Ranking.new(ranking_params)
+    #
+    # respond_to do |format|
+    #   if @ranking.save
+    #     format.html { redirect_to @ranking, notice: 'Ranking was successfully created.' }
+    #     format.json { render :show, status: :created, location: @ranking }
+    #   else
+    #     format.html { render :new }
+    #     format.json { render json: @ranking.errors, status: :unprocessable_entity }
+    #   end
+    # end
+
+    @times.ids.each do |id|
+      unless params.keys.include? id.to_s
+        flash[:error] = "Error: please fill in an option for each time."
+        redirect_to edit_project_participant_ranking_path(:secret_id => @participant.secret_id) and return
+      end
+    end
+
+    @times.ids.each do |id|
+      rank_num = params[id.to_s].to_i
+      new_rank = Ranking.new(:id => @ranking.id, :rank => rank_num, :participant_id => @participant.id, :project_time_id => id)
+      existing_rank = Ranking.find_by(:id => @ranking.id, :project_time_id => id)
+
+      if existing_rank
+        existing_rank.destroy
+      end
+      new_rank.save!
+    end
+
+    redirect_to end_project_participant_ranking_path
+  end
+
+  # PATCH/PUT /rankings/1
+  # PATCH/PUT /rankings/1.json
+  def update
+    @times.ids.each do |id|
+      unless params.keys.include? id.to_s
+        flash[:error] = "Error: please fill in an option for each time."
+        redirect_to edit_project_participant_ranking_path(:secret_id => @participant.secret_id) and return
+      end
+    end
+
+    @times.ids.each do |id|
+      rank_num = params[id.to_s].to_i
+      new_ranking = Ranking.new(:rank => rank_num, :participant_id => @participant.id, :project_time_id => id)
+      existing_ranking = Ranking.find_by(:participant_id => @participant.id, :project_time_id => id)
+
+      if existing_ranking
+        Ranking.find_by(:participant_id => @participant.id, :project_time_id => id).update(
+            :rank => rank_num, :participant_id => @participant.id, :project_time_id => id)
+      else
+        new_ranking.save!
+      end
+    end
+
```

Comment:
I think we're missing the feature to update the participant's `last_responded` column when we click submit?